client/components/course/Sidebar/Relationship.vue

```diff
@@ -55,8 +59,19 @@ export default {
         const lineage = this.getLineage(this.activity);
         conds.push(it => !includes(lineage, it));
       }
+      if (this.allowedTypes && this.allowedTypes.length) {
```

Comment:
I'm all for using empty array as default but this implementation actually matches one of `SelectElement` and others throughout the system where defaults for whitelist arrays are nulls...

Reply:
Regarding `isEmpty` point I would go into opposite direction by eliminating it completely because it is used for checking of `this.options` length which is guaranteed to be an array. So either inline that length check there or implement _one line_ `isEmpty` helper instead of pulling complex lodash implementation sounds more practical to me.